core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -84,9 +76,111 @@ abstract class InterBrokerSendThread(name: String,
     }
   }
 
-  def wakeup(): Unit = networkClient.wakeup()
+  private def sendRequests(now: Long): Long = {
+    var pollTimeout = Long.MaxValue
+    val iterator = unsentRequests.iterator()
+    while (iterator.hasNext) {
+      val entry = iterator.next
+      val node = entry.getKey
+      val requestIterator = entry.getValue.iterator()
+      while (requestIterator.hasNext) {
+        val request = requestIterator.next
+        if (networkClient.ready(node, now)) {
+          networkClient.send(request, now)
+          requestIterator.remove()
+        } else
+          pollTimeout = Math.min(pollTimeout, networkClient.connectionDelay(node, now))
+      }
+    }
+    pollTimeout
+  }
+
+  private def checkDisconnects(now: Long): Unit = {
```

Comment:
Would be good to have a test case which hits this path. I noticed that `InterBrokerSendThread` depends on `NettworkClient` directly. Maybe we could change that to use `KafkaClient` instead so that we can use `MockClient` for easier testing?

Reply:
@hachikuji Looking at this more closely, wouldn't the existing test `shouldCallCompletionHandlerWithDisconnectedResponseWhenNodeNotReady` cover this? In the latest commit I made a minor change to verify completion with disconnect. Would that be a proper test for this path? If so, should I still look into switching to `MockClient`? Thanks.